tests/automation-framework/rancherclient/config.go

```diff
@@ -0,0 +1,94 @@
+package rancherclient
+
+type Config struct {
+	CattleTestURL     string `json:"cattleTestURL"`
+	AdminToken        string `json:"adminToken"`
+	UserToken         string `json:"userToken,omitempty"`
+	CNI               string `json:"cni" default:"calico"`
+	KubernetesVersion string `json:"kubernetesVersion"`
+	NodeRoles         string `json:"nodeRoles,omitempty"`
+	DefaultNamespace  string `default:"fleet-default"`
+	Insecure          *bool  `json:"insecure" default:"true"`
```

Comment:
this doesn't need to be a pointer, we can pass it's pointer to the downstream clients if they need one but it complicates setting the configuration.

Reply:
I have this as a pointer, in the case for validation and `Insecure` for whatever reason is not in the json file. That way `Insecure` can be defaulted to to true, if it's not present in the json file.

Reply:
ah that makes sense